### PR TITLE
Archaeology

### PR DESCRIPTION
**Pull request by ThatDamnPineapple**
This PR adds a new archaeology system, which allows for buried artifacts to generate under tiles, as well as the following artifacts:
-Various desert fossils
-Lava charm artifact
-Pirate treasure artifact

Follows https://trello.com/c/ZfTWwmlI/20-archaeology-backend-8 to a T

Nothing left to be done except more artifacts and potentially some updates to optimization?

### CODE REVIEW

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ScalarVector1) [inline Content/Archaeology/ArchaeologyManager.cs:21]: Lets be a bit careful while naming things -- XManager is never used in any other place in the mod, ArchaeologySystem or ArchaeologyHandler would be more appropriate

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ScalarVector1) [inline Content/Archaeology/ArchaeologyManager.cs:37]: single line conditionals should be inlined

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ScalarVector1) [inline Content/Archaeology/ArchaeologyManager.cs:48]: single line conditionals should be inlined

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ScalarVector1) [inline Content/Archaeology/ArchaeologyManager.cs:61]: It would probably be a good idea to cache these instead of scanning through to find them every draw tick

File: Content/Archaeology/Artifact.cs
Comment (ScalarVector1) [inline Content/Archaeology/Artifact.cs:25]: World coordiantes are generally stored in Point16 variables -- should we be using the same here?

File: Content/Archaeology/Artifact.cs
Comment (ScalarVector1) [inline Content/Archaeology/Artifact.cs:39]: lets use summary comments here so documentation appears when hovering over this in VS

File: Content/Archaeology/ArtifactItemProj.cs
Comment (ScalarVector1) [inline Content/Archaeology/ArtifactItemProj.cs:106]: Dont use arrow notation for this method

File: Content/Archaeology/BuriedArtifacts/Artifacts.ArtifactTypes.cs
Comment (ScalarVector1) [inline Content/Archaeology/BuriedArtifacts/Artifacts.ArtifactTypes.cs:16]: Should we really be creating classes like this? Can we not have a load method similar to Mod.AddContent?

File: Content/Archaeology/BuriedArtifacts/Artifacts.DesertArtifacts.cs
Comment (ScalarVector1) [inline Content/Archaeology/BuriedArtifacts/Artifacts.DesertArtifacts.cs:16]: Similar sentiment to the other comment on class creation -- A method to load this content with parameters for basic configuration like spawn chance and size would be much  cleamer

File: Content/Dusts/ArtifactSparkles/ArtifactSparkles.cs
Comment (ScalarVector1) [inline Content/Dusts/ArtifactSparkles/ArtifactSparkles.cs:55]: Is there a benefit to this over making the sprite greyscale and using the color parameter?

File: Content/Items/BuriedArtifacts/Artifacts.CommonDesertArtifactItems.cs
Comment (ScalarVector1) [inline Content/Items/BuriedArtifacts/Artifacts.CommonDesertArtifactItems.cs:6]: These should probably get autoloaded with the artifacts isntead of being defined seperately -- this could probably included in that parameterized loading method

File: Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs
Comment (ScalarVector1) [inline Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs:10]: missing linebreak

File: Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs
Comment (ScalarVector1) [inline Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs:27]: dont use arrow notation mid-class

File: Content/Items/Misc/ArchaeologistsMap.cs
Comment (ScalarVector1) [inline Content/Items/Misc/ArchaeologistsMap.cs:54]: single line conditionals should be inlined

**Review by ScalarVector1 — CHANGES_REQUESTED** [review]
Most of this looks really nice, my main concern is with how artifacts are loaded. Using OOP for artifacts without any special effects seems rather goofy, along with items having to be loaded seperately from the actual artifacts themselves.

Additionally there are a few minor styling issues that should be resolved before this is merged.

File: Content/Archaeology/Artifact.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/Artifact.cs:25]: By world coordinates I meant the position on the same grid NPCs and projectiles have their positions on.

File: Content/Dusts/ArtifactSparkles/ArtifactSparkles.cs
Comment (ThatDamnPineapple) [reply Content/Dusts/ArtifactSparkles/ArtifactSparkles.cs:55]: hueshifting

File: Content/Items/BuriedArtifacts/Artifacts.CommonDesertArtifactItems.cs
Comment (ThatDamnPineapple) [reply Content/Items/BuriedArtifacts/Artifacts.CommonDesertArtifactItems.cs:6]: Except they're going to all have unique/special behaviors in the future

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/ArchaeologyManager.cs:21]: Fixed

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/ArchaeologyManager.cs:37]: Fixed

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/ArchaeologyManager.cs:48]: Fixed

File: Content/Archaeology/ArchaeologyManager.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/ArchaeologyManager.cs:61]: Done

File: Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs
Comment (ThatDamnPineapple) [reply Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs:10]: Fixed

File: Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs
Comment (ThatDamnPineapple) [reply Content/Items/BuriedArtifacts/Artifacts.PirateChestArtifact.cs:27]: Didn't know this was a thing, fixed

File: Content/Items/Misc/ArchaeologistsMap.cs
Comment (ThatDamnPineapple) [reply Content/Items/Misc/ArchaeologistsMap.cs:54]: Fixed

File: Content/Archaeology/Artifact.cs
Comment (ThatDamnPineapple) [reply Content/Archaeology/Artifact.cs]: Done

File: Content/Archaeology/Artifact.cs
Comment (ScalarVector1) [inline Content/Archaeology/Artifact.cs:136]: missing linebreak

**Review by ScalarVector1 — APPROVED** [review]
Looks good apart from a few tiny style nits, should be good to merge after these changes. Additionally, we should probably put down a note somewhere to change the loading mechanism for these once tmod updates to allow overriding TileEntity name property.

**Pull request merged**
